src/include/storage/data_table.h

```diff
@@ -74,6 +74,8 @@ class DataTable {
   void Rollback(timestamp_t txn_id, TupleSlot slot);
 
  private:
+  friend class GarbageCollector;
```

Comment:
No need to fix it in this PR, but we should be more consistent with this and the rollback function from transactions (whether to write a function in DataTable or make friend class)

Reply:
Friend class for GC definitely makes sense since it’s in the same namespace and abstraction layer and we don’t want to expose anything extra from the DT that GC might need to other layers.

It gets weird with CC. Friend class might make sense there too if we’re trying encapsulate everything in the backend (storage and CC) from the ref of the system — I.e. not expose something like Rollback to the execution layer.